### ISSUE TITLE
Wrap: Allow annotations parameter to accept specific fields

### ISSUE THREAD

**Issue by daniel-j-h** [Project-OSRM/node-osrm#298]
The `annotations` parameter changed form a `bool` to specific annotation types.
We need to wrap this in `node-osrm`, too.

Upstream:

- Issue https://github.com/Project-OSRM/osrm-backend/issues/3563
- Pull Request https://github.com/Project-OSRM/osrm-backend/pull/3628

Tasks
- [ ] `node_osrm.cpp` adapt comments
- [ ] `node_osrm_support.hpp` parse from Javascript-land
- [ ] adapt `docs/api.md`

[Here's](https://github.com/Project-OSRM/node-osrm/blob/139b7ebf082996478139d372c981277eb9370c16/src/node_osrm_support.hpp#L420-L432) the entry point where we throw an exception currently.

cc @karenzshea

**Comment by karenzshea** [comment]
I'll hit this after the `speed` param has also been added as an annotations option.

**Comment by daniel-j-h** [comment]
In order to not break the ABI here's what we have to do:
- support the old boolean parameter
- allow new string parameters

This is Javascript-land, we can do this by means of checking `v8::Value::IsBoolean` (old behavior) and `v8::Value::IsString()` (new behavior).

**Comment by chaupow** [comment]
How should the API look like? 
currently I was thinking of supporting following params:
```
var options = {
    annotations: 'true'
};

var options = {
    annotations: true
};

var options = {
    annotations: ['distance', 'weight', 'node']
};
```

does this make sense?

@karenzshea @TheMarex @daniel-j-h @freenerd

**Comment by TheMarex** [comment]
@chaupow I would suggest not allowing string values but only boolean and array of strings to keep the API surface small.

**Comment by chaupow** [comment]
So shall we accept `bool` or an `array of string`?

Then, I will accept
```
var options = {
    annotations: true
};
var options = {
    annotations: ['distance', 'weight', 'node']
};
var options = {
    annotations: ['distance']
};
```
 I will **not** accept
```
var options = {
    annotations: 'distance'
};
var options = {
    annotations: [true]
};
```

?

**Comment by TheMarex** [comment]
@chaupow exactly.

**Comment by chaupow** [comment]
merged #299
